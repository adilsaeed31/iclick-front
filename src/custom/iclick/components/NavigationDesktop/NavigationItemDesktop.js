import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import { inject } from "mobx-react";
import Link from "custom/iclick/components/Link";

@inject("routingStore")
class NavigationItemDesktop extends Component {
  static propTypes = {
    navItem: PropTypes.object,
    routingStore: PropTypes.object
  }

  static defaultProps = {
    navItem: {},
    routingStore: {}
  }

  state = {
    show: "megamenu-container"
  }

  linkPath = (providedNavItem) => {
    const { navItem, routingStore } = this.props;

    const currentNavItem = providedNavItem || navItem;

    return routingStore.queryString !== ""
      ? `/tag/${currentNavItem.slug}?${routingStore.queryString}`
      : `/tag/${currentNavItem.slug}`;
  }

  get hasSubNavItems() {
    // const {
    //   navItem: { subTags }
    // } = this.props;
    const subTags=this.props.navItem.items;
    return Array.isArray(subTags) && subTags.length > 0;
  }
  
  renderPopover() {
    const subMenuImg={
      width: "100%",
      minHeight: "-webkit-fill-available",
      overflow: "hidden",
      display: "inline-block"
    }
    // const {
    //   navItem: { subTags }
    // } = this.props;
    //const subTags = this.props.navItem.items.map(x=>x.navigationItem.data);
    const subTags = this.props.navItem.items;
    if (subTags) {
      return (
        <div className="megamenu" style={{height:"50vh",overflow:"hidden"}}>
          <div className="row">
            <div className="col-lg-8">
              <div className="row">
                {subTags.map((navSubItems,index) => (
                  <div className="col-lg-3" key={index}>
                    <div className="menu-title">
                        {console.log("navSubItems.navigationItem.data.url", navSubItems.navigationItem.data ? navSubItems.navigationItem.data.url : navSubItems.navigationItem.data)}
                      <Link key={index} route={navSubItems.navigationItem.data.url}>{navSubItems.navigationItem.data.contentForLanguage}</Link>
                    </div>

                    
                  </div>
                ))}
              </div>
            </div>
            <div className="col-lg-4">
              <div className="banner">
                <a href="/" style={subMenuImg}>
                  <img style={{height:"100%",width:"auto",}} src="/static/images/menu-banner.jpg" alt="Menu banner" className="product-promo" />
                </a>
              </div>
            </div>
          </div>
        </div>
      );
    }

    return null;
  }

  handleMouseOver = () => {
    this.setState({ show: "megamenu-container show" });
  }

  handleMouseOut = () => {
    this.setState({ show: "megamenu-container" });
  }

  render() {
    const { navItem } = this.props;
    const { show } = this.state;
    return (
      <li onMouseOver={this.handleMouseOver} className={show} onMouseOut={this.handleMouseOut}>
         <Link className={this.hasSubNavItems ? "sf-with-ul" : ""} route={this.linkPath(navItem)}>
          {navItem.navigationItem.data.contentForLanguage}
          </Link>
        {this.hasSubNavItems && this.renderPopover()}
      </li>
    );
  }
}

export default NavigationItemDesktop;
