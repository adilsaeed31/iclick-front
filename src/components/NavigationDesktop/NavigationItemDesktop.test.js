import React from "react";
import renderer from "react-test-renderer";
import NavigationItemDesktop from "./NavigationItemDesktop";

const mockNavItem = {
  items: [],
  navigationItem: {
    data: {
      classNames: null,
      content: [{
        language: "en",
        value: "NavItem"
      }],
      isUrlRelative: true,
      shouldOpenInNewWindow: false,
      url: "/nav-item"
    }
  }
};

const mockNavItemWithSubItem = {
  items: [{
    items: [],
    navigationItem: {
      data: {
        classNames: null,
        content: [{
          language: "en",
          value: "NavSubItem"
        }],
        isUrlRelative: true,
        shouldOpenInNewWindow: false,
        url: "/nav-sub-item"
      }
    }
  }],
  navigationItem: {
    data: {
      classNames: null,
      content: [{
        language: "en",
        value: "NavItem"
      }],
      isUrlRelative: true,
      shouldOpenInNewWindow: false,
      url: "/nav-item"
    }
  }
};

const mockNavItemWithSubItems = {
  items: [{
    items: [{
      items: [],
      navigationItem: {
        data: {
          classNames: null,
          content: [{
            language: "en",
            value: "NavSubSubItem"
          }],
          isUrlRelative: true,
          shouldOpenInNewWindow: false,
          url: "/nav-sub-sub-item"
        }
      }
    }],
    navigationItem: {
      data: {
        classNames: null,
        content: [{
          language: "en",
          value: "NavSubItem"
        }],
        isUrlRelative: true,
        shouldOpenInNewWindow: false,
        url: "/nav-sub-item"
      }
    }
  }],
  navigationItem: {
    data: {
      classNames: null,
      content: [{
        language: "en",
        value: "NavItem"
      }],
      isUrlRelative: true,
      shouldOpenInNewWindow: false,
      url: "/nav-item"
    }
  }
};

const mockRoutingStore = {
  queryString: ""
};

test("renders with 1 level of tags", () => {
  const component = renderer.create(<NavigationItemDesktop navItem={mockNavItem} routingStore={mockRoutingStore} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders with 2 levels of tags", () => {
  const component = renderer.create(<NavigationItemDesktop navItem={mockNavItemWithSubItem} routingStore={mockRoutingStore} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders with 3 levels of tags", () => {
  const component = renderer.create(<NavigationItemDesktop navItem={mockNavItemWithSubItems} routingStore={mockRoutingStore} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
