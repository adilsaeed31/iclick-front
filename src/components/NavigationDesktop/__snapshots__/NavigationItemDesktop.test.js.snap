// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with 1 level of tags 1`] = `
<a
  className="MuiButtonBase-root-32 MuiButton-root-6 MuiButton-text-8 MuiButton-flat-11 MuiButton-colorInherit-27 SkNavigationItemDesktop-primaryNavItem-5"
  href="/nav-item"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="MuiButton-label-7"
  />
  <span
    className="MuiTouchRipple-root-35"
  />
</a>
`;

exports[`renders with 2 levels of tags 1`] = `
<a
  className="MuiButtonBase-root-32 MuiButton-root-6 MuiButton-text-8 MuiButton-flat-11 MuiButton-colorInherit-27 SkNavigationItemDesktop-primaryNavItem-5"
  href="/nav-item"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="MuiButton-label-7"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-42"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-35"
  />
</a>
`;

exports[`renders with 3 levels of tags 1`] = `
<a
  className="MuiButtonBase-root-32 MuiButton-root-6 MuiButton-text-8 MuiButton-flat-11 MuiButton-colorInherit-27 SkNavigationItemDesktop-primaryNavItem-5"
  href="/nav-item"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="MuiButton-label-7"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-42"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-35"
  />
</a>
`;
